app/models/solr_document.rb

```diff
@@ -164,13 +164,11 @@ def bound_with?
   def holdings_all_display
     holdings = JSON.parse(self["holdings_1display"] || '{}')
     return holdings if host_id.nil?
-
     # Fetch and append the holdings from the host record
-    @holdings_all ||= begin
-      host_doc = doc_by_id(host_id)
-      host_holdings = JSON.parse(host_doc&.dig("holdings_1display") || '{}')
-      holdings.merge(host_holdings)
+    host_id.each do |id|
+      holdings.merge!(JSON.parse(doc_by_id(id)&.dig("holdings_1display") || '{}'))
```

Comment:
This is a bit complex to all be on one line.  The call out to `doc_by_id(id)` got lost on me and I had a hard time realizing it was actually querying solr at that point and then getting the holdings display out of the document and then parsing it.  For readability it might make sense to utilize some intermediate variables like:
```
  host_solr_document = doc_by_id(id)
  host_holdings = host_solr_document&.dig("holdings_1display")
  holdings.merge! (JSON.parse(host_holdings)) if host_holdings.present? #do not merge an empty holding
```